node/src/components/small_network/outgoing.rs

```diff
@@ -426,6 +426,14 @@ where
         self.routes.keys().cloned()
     }
 
+    /// Iterates over all outgoing connections.
+    #[allow(clippy::needless_lifetimes)]
```

Comment:
Curiosity box: Why do you need this annotation? Would `clippy` complain that you can remove the lifetime but it wouldn't actually work?

Reply:
It compiles for me if I just remove all the lifetimes.